samples/drivers/espi/src/main.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <soc.h>
+#include <gpio.h>
+#include <espi.h>
+
+#ifdef CONFIG_ESPI_GPIO_DEV_NEEDED
+static struct device *gpio_dev;
+#endif
+
+static struct device *espi_dev;
+
+static struct espi_callback espi_bus_cb;
+static struct espi_callback vw_rdy_cb;
+static struct espi_callback vw_cb;
+static struct espi_callback p80_cb;
+
+/* eSPI bus event handler */
+static void espi_reset_handler(struct device *dev,
+				struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_RESET) {
+		printk("\neSPI BUS reset %d", event.evt_data);
+	}
+}
+
+/* eSPI logical channels enable/disable event handler */
+static void espi_ch_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_EVENT_CHANNEL_READY) {
+		if (event.evt_details == ESPI_CHANNEL_VWIRE) {
+			printk("\nVW channel is ready\n");
+		}
+	}
+}
+
+/* eSPI vwire received event handler */
+static void vwire_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	if (event.evt_type == ESPI_BUS_EVENT_VWIRE_RECEIVED) {
+		if (event.evt_details == ESPI_VWIRE_SIGNAL_PLTRST) {
+			printk("\nPLT_RST changed %d\n", event.evt_data);
+		}
+	}
+}
+
+/* eSPI peripheral channel notifications handler */
+static void periph_handler(struct device *dev, struct espi_callback *cb,
+				struct espi_event event)
+{
+	u8_t peripheral;
+
+	if (event.evt_type == ESPI_BUS_PERIPHERAL_NOTIFICATION) {
+		peripheral = event.evt_details & 0x00FF;
+		switch (peripheral) {
+		case ESPI_PERIPHERAL_DEBUG_PORT80:
+			printk("Postcode %x\n", event.evt_data);
+			break;
+		case ESPI_PERIPHERAL_HOST_IO:
+			printk("ACPI %x\n", event.evt_data);
+			break;
+		default:
+			printk("\n%s periph 0x%x [%x]\n", __func__, peripheral,
+				event.evt_data);
+		}
+	}
+}
+
+/* Indicate to eSPI master only single line, 20MHz frequency and
```

Comment:
Indicate to eSPI master the following configuration : 
Single Line, 20MHz frequency and logical channel 0.